scripts/utils/taco_rewards.js

```diff
@@ -232,76 +233,132 @@ async function getPotentialRewards(startPeriodTimestamp, endPeriodTimestamp) {
 
   const rewards = {};
   Object.keys(confirmedAuthHistories).forEach((stProv) => {
-    // Take the first event: this is, the event whose timestamp or
-    // opConfirmedTimestamp (the oldest of both) is closer to (and preferably
-    // lower than) start period time
-    const firstEvent = confirmedAuthHistories[stProv].reduce((acc, cur) => {
-      const accTimestamp = Number(acc.timestamp);
-      const curTimestamp = Number(cur.timestamp);
+    // sorting the events by timestamp in case they are not sorted yet
+    const sortedEvents = confirmedAuthHistories[stProv].sort(
+      (a, b) => Number(a.timestamp) - Number(b.timestamp)
+    );
+
+    let accEligibleAmount = BigNumber(0);
+    let accDecreaseRequestedAmount = BigNumber(0);
+
+    // based on these auth events, let's create "authorization epochs"
+    // - each authorization epoch is defined by an authorization change event
+    // - each authorization epoch has a eligible amount and duration
+    // for example, if a stake decreased the full authorization in the half
+    // of the month, this stake will have 2 epochs: one with full amount
+    // and duration of 15 days, and another epoch with 0 amount and duration
+    // of 15 days
+
+    // note that we will calculate authorization epochs previous to this rewards
+    // period, but we will filter them later
+    const epochs = [];
+    // let's generate the epochs and calculate the rewards eligible amount based
+    // on the auth events history (queried to the subgraph)
+    for (let i = 0; i < sortedEvents.length; i++) {
+      const eventType = sortedEvents[i].eventType;
+      bigNumberEventAmount = BigNumber(sortedEvents[i].eventAmount);
+      if (eventType === "AuthorizationIncreased") {
+        // on increase, add the amount to the rewards eligible amount
+        accEligibleAmount = accEligibleAmount.plus(bigNumberEventAmount);
+      } else if (eventType === "AuthorizationDecreaseRequested") {
+        // on decrease REQUESTED, deduct the amount of rewards eligible amount
+        accEligibleAmount = accEligibleAmount.minus(bigNumberEventAmount);
+        accDecreaseRequestedAmount =
+          accDecreaseRequestedAmount.plus(bigNumberEventAmount);
+      } else if (
+        eventType === "AuthorizationDecreaseApproved" ||
+        eventType === "AuthorizationInvoluntaryDecreased"
```

Comment:
would this 2 events used for all stakers or just who doesn't have "AuthorizationDecreaseRequested"?